api/services/dataset_service.py

```diff
@@ -1521,37 +1521,26 @@ def update_segment(cls, args: dict, segment: DocumentSegment, document: Document
                     disable_segment_from_index_task.delay(segment.id)
                     return segment
         if not segment.enabled:
-            if "enabled" in args and args["enabled"] is not None:
-                if not args["enabled"]:
+            if segment_update_entity.enabled is not None:
+                if not segment_update_entity.enabled:
```

Comment:
I have to say it's in the right way to guarantee the possible null value, although it's still very tricky in usage.